### ISSUE TITLE
dotnet cli created projects don't work in VS2017 15.3.3.

### ISSUE THREAD

**Issue by davkean** [dotnet/project-system#2812]
_From @codefactorydevelopment on September 9, 2017 13:58_

## Steps to reproduce

Create 2 projects:

 1. `dotnet new webapi` (name: webapiProject)
 2. `dotnet new xunit` (name: testProject)

After that, on `testProject` project:
`dotnet add reference webapiProject`

Now, when opening testProject.csproj in `VS2017 15.3.3.`

## Expected  behavior
Expected to see `testProject` loaded with resolved dependency to `webapiProject`

## Actual behavior

When opening testProj.csproj in VS2017 15.3.3. the following Nuget **NU1105** error occurs:

*Severity	Code	Description	Project	File	Line	Suppression State
Error	NU1105	Unable to find project information for 'webapiProject'. The project file may be invalid or missing targets required for restore.*	

Dependencies & build work as expected when done from `dotnet cli`.

## Environment data
`dotnet --info` output:

.NET Command Line Tools (2.0.0)

Product Information:
 Version:            2.0.0
 Commit SHA-1 hash:  cdcd1928c9

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.14393
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.0.0\

Microsoft .NET Core Shared Framework Host

  Version  : 2.0.0
  Build    : e8b8861ac7faf042c87a5c2f9f2d04c98b69f28d

_Copied from original issue: dotnet/cli#7623_

**Comment by davkean** [comment]
_From @nguerrera on September 9, 2017 22:35_

@emgarten

**Comment by davkean** [comment]
_From @emgarten on September 10, 2017 0:44_

This can happen if the project system does not nominate a project. 

Does the error continue if you wait?
Does activitylog.xml from VS contain any additional errors on why the project failed?
Are you consistently getting this error?

**Comment by davkean** [comment]
_From @codefactorydevelopment on September 10, 2017 5:55_

hi @emgarten 

 1. Error is there regardless of how many times i try or how long i wait.
 2. Just checked, there is nothing in activitylog.xml
 3. I've tried this scenario a few times and always got that error.

Could i ask, what does 'nominating a project' exactly mean? Not sure i've ever heard that term.

**Comment by davkean** [comment]
_From @nguerrera on September 10, 2017 14:35_

@davkean Should I move this to project system?

**Comment by davkean** [comment]
_From @nguerrera on September 11, 2017 0:32_

@codefactorydevelopment Are the two project files in the same directory?

**Comment by davkean** [comment]
_From @codefactorydevelopment on September 11, 2017 2:38_

@nguerrera

no, they are each in their own folder:

-parentFolder
|---webapiProject/webapiProject.csproj
|---testProject/testProject.csproj

**Comment by davkean** [comment]
@natidea Can you root cause this?

**Comment by dasMulli** [comment]
@codefactorydevelopment are the two projects in a solution or are you just opening the test project on VS without a solution file?

**Comment by codefactorydevelopment** [comment]
@dasMulli

they are not in a solution. i am opening testProject in VS without solution file.

**Comment by dasMulli** [comment]
Then this likely is a duplicate of https://github.com/dotnet/project-system/issues/2759 (restore doesn't work for project references not included in a solution)

**Comment by codefactorydevelopment** [comment]
@dasMulli

oh yes, looks like that. thanks for pointing that out!

so for now seems like adding to solution is simplest workaround.

**Comment by dsplaisted** [comment]
It seems reasonable to me not to fix #2759 (ie building projects in VS that aren't part of the solution), as @davkean suggests on that bug.  However, we should improve the error experience here.  Rather than:

    NU1105	Unable to find project information for 'webapiProject'. The project file may be invalid or missing targets required for restore.

The error message should tell the user that the referenced project needs to be added to the solution.  I'm not sure if that needs to be fixed in NuGet, or in the project system, or both.  @emgarten

**Comment by emgarten** [comment]
@dsplaisted improving this error message for visual studio scenarios is tracked here: https://github.com/NuGet/Home/issues/5350

**Comment by natidea** [comment]
Closing this as a duplicate of https://github.com/dotnet/project-system/issues/2759

**Comment by julielerman** [comment]
I got this error message when creating a new solution, copying two projects from elsewhere into the solution's folder, then back in VS2017 15.2, adding both projects to the solution. When building, I got the error on multiple attempts. Closing and re-opening VS2017 made the problem disappear. Unfortunately, this seems to be the solution to a variety of weird problems that I've had over time which is why I do it before wasting too much time scratching my head.
